src/stream/src/executor/top_n/top_n_plain.rs

```diff
@@ -115,29 +104,21 @@ impl<S: StateStore, const WITH_TIES: bool> InnerTopNExecutor<S, WITH_TIES> {
     /// into `CacheKey`.
     #[allow(clippy::too_many_arguments)]
     pub fn new(
-        input_info: ExecutorInfo,
+        info: ExecutorInfo,
         storage_key: Vec<ColumnOrder>,
         offset_and_limit: (usize, usize),
         order_by: Vec<ColumnOrder>,
-        executor_id: u64,
         state_table: StateTable<S>,
     ) -> StreamResult<Self> {
-        let ExecutorInfo {
-            pk_indices, schema, ..
```

Comment:
Later we may completely more `ExecutorInfo` out of `Executor` to get rid of this kind of bug. While this may be related to the `Update` issue https://github.com/risingwavelabs/risingwave/issues/12539.